array.c

```diff
@@ -7843,13 +7843,22 @@ rb_ary_deconstruct(VALUE ary)
  *
  *  == Creating Arrays
  *
- *  A new array can be created by using the literal constructor
- *  <code>[]</code>.  Arrays can contain different types of objects.  For
+ *  You can create an \Array object explicitly with:
+ *
+ *  - Global method {Array}[Kernel.html#method-i-Array].
+ *  - A {array literal}[doc/syntax/literals_rdoc.html#label-Array+Literals].
```

Comment:
The global methods are used for typecasting existing data to the expected type, they are not used for creating new instances (they accept an argument).  If you mention the global methods at all, it should be under the section that discusses `Many Ruby methods return #{type} objects`.  Also, please refer to them as Kernel methods, not global methods, since they aren't truly global (cannot use them in a BasicObject).

For the objects being discussed where you can use `#{class}.new` (e.g. `Array#new`), it should be listed directly after the literal form, since that is the next most common way of creating a new instance.

FWIW, I prefer the previous documentation even after fixing these issues.  I don't think mentioning the `to_a`/`to_ary`  methods (or similar methods for other classes) is important.

Reply:
> FWIW, I prefer the previous documentation even after fixing these issues.

Do you mean this to cover this entire PR?

Reply:
No, only the parts of the PR where you are adding documentation about global methods and `to_a`/`to_ary` and similar methods.

Reply:
Done (I think).